src/typed.js

```diff
@@ -295,3 +295,23 @@ Typed.Number.Range = (from, to=+Infinity, defaultValue) =>
 
     return value
   }, defaultValue)
+
+Typed.Enum = (values, defaultValue) =>{
+  if (!(values instanceof Array)) {
+    throw TypeError(`"${values}" is not an array`)
+  }
+
+  const map = {}
+  values.forEach((val) => {
+    map[val] = val
+  });
+  const validValues = values.join(', ');
+
+
+  return Typed(`Enum(${validValues})`, value => {
+    if (!map[value] && value !== undefined){
```

Comment:
I don't think the `value !== undefined` check should be there - if user want's an optional value shouldn't they choose to use `Maybe`?